components/skus/browser/rs/lib/src/storage/kv.rs

```diff
@@ -210,6 +319,97 @@ where
         store.set_state(&state)
     }
 
+    #[instrument]
+    async fn clear_time_limited_v2_creds(
+        &self,
+        item_id: &str,
+    ) -> Result<(), InternalError> {
+
+        let mut store = self.get_store()?;
+        let mut state: KVState = store.get_state()?;
+
+        if let Some(credentials) = state.credentials.as_mut() {
+            if let Some(item_credentials) = credentials.items.get_mut(item_id) {
+                match item_credentials {
+                    Credentials::TimeLimitedV2(item_credentials) => {
+                        // clear out the item_credential tokens, they have been signed
+                        item_credentials.creds = Vec::new()
+                    }
+                    _ => {
+                        return Err(InternalError::StorageWriteFailed(
+                            "Item is not time limited v2".to_string(),
+                        ));
+                    }
+                }
+                return store.set_state(&state);
+            }
+        }
+        Err(InternalError::StorageWriteFailed("Item tokens were not found".to_string()))
+    }
+
+    #[instrument]
+    async fn append_time_limited_v2_item_creds(
```

Comment:
could use a bit of naming tweak to differentiate from creds methods above, maybe `append_time_limited_v2_item_unblinded_creds`?